src/NHibernate/Dialect/Function/ISQLFunctionExtended.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using NHibernate.Engine;
+using NHibernate.Type;
+
+namespace NHibernate.Dialect.Function
+{
+	// 6.0 TODO: Merge into ISQLFunction
+	internal interface ISQLFunctionExtended : ISQLFunction
+	{
+		/// <summary>
+		/// The function name or <see langword="null"/> when multiple functions/operators/statements are used.
+		/// </summary>
+		string FunctionName { get; }
```

Comment:
I think this should be renamed to `Name` to reduce need of obsoleting things.

Reply:
Why not, renamed in #2359.